### PR TITLE
Parser e2e tests

### PR DESCRIPTION
**Pull request by ggawryal**
Tests are failing, but after merging with https://github.com/TCS-Compilers/compiler/pull/35 and https://github.com/TCS-Compilers/compiler/pull/39 they should pass

### CODE REVIEW

**Review by rkilar — COMMENTED** [review]
LGTM. I've confirmed that the tests pass with current parser PRs. Once those are merged and tests pass it will get approved.

File: app/src/test/kotlin/compiler/parser/integration_tests/ExpressionParsingTest.kt [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import kotlin.test.assertEquals
+
+class ExpressionParsingTest : ParserTest() {
+    private val mulExpr = 'A'
+    private val addExpr = 'B'
+    private val numExpr = 'C'
+
+    private val addToMul = Production(addExpr, RegexParser.parseStringToRegex("$mulExpr(\\+$mulExpr)*"))
+    private val mulToNum = Production(mulExpr, RegexParser.parseStringToRegex("$numExpr(\\*$numExpr)*"))
+    private val numParentheses = Production(numExpr, RegexParser.parseStringToRegex("\\(${addExpr}\\)"))
```
Comment (rkilar) [inline app/src/test/kotlin/compiler/parser/integration_tests/ExpressionParsingTest.kt:19]: RegexParser was renamed to LexerRegexParser

File: app/src/test/kotlin/compiler/parser/integration_tests/NonLL1Grammar.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import compiler.parser.grammar.Grammar
+import compiler.parser.grammar.Production
+import org.junit.Test
+import kotlin.test.assertFailsWith
+
+class NonLL1Grammar : ParserTest() {
+    private val aToB = Production('A', RegexParser.parseStringToRegex("B"))
+    private val aToC = Production('A', RegexParser.parseStringToRegex("C"))
+    private val bToab = Production('B', RegexParser.parseStringToRegex("ab"))
+    private val cToac = Production('C', RegexParser.parseStringToRegex("ac"))
```
Comment (rkilar) [inline app/src/test/kotlin/compiler/parser/integration_tests/NonLL1Grammar.kt:14]: Same here

**Review by rkilar — CHANGES_REQUESTED** [review]

**Comment by ggawryal** [comment]
Fixed

**Review by rkilar — APPROVED** [review]

**Pull request merged**